public/js/report.js

```diff
@@ -117,22 +117,40 @@ var Report = React.createClass({
     updateState: function(responseJSON) {
         var queries = responseJSON.queries;
         responseJSON.veryBig = queries.length > 250;
-        responseJSON.queries = queries.splice(0, 50);
+        responseJSON.queries = queries.splice(0, 1);
         this.setState(responseJSON);
-
+        var start = Date.now();
         // Render results for remaining queries.
         var update = function () {
+            var hit_num = 0, query_num = 0;
+            queries.some(function (query) {
+                hit_num += query.hits.length;
+                query_num++;
+                if (hit_num > 500) {
+                    query_num -= 1;
+                    return query_num;
+                } else if (hit_num == 500) {
+                    return query_num;
+                } else if (query_num == queries.length) {
```

Comment:
The last `else if` is unnecessary: if it is the last update cycle and hit_num is less than 500 `queries.some` will iterate to completion and return `false`, which is fine because its return value is not used anywhere.

And the remainder can be simplified to:
```
if (hit_num > 500) query_num--;
return hit_num >= 500;
```